src/tools/tidy/src/features.rs

```diff
@@ -82,6 +82,7 @@ pub fn collect_lib_features(base_src_path: &Path) -> Features {
 
 pub fn check(
     src_path: &Path,
+    tests_path: &Path,
     compiler_path: &Path,
     lib_path: &Path,
```

Comment:
maybe just pass in `root_path` to avoid blowing up the number of arguments? but this PR is already quite large, I don't mind making that a follow-up.

Reply:
It would make another thing dependant on the place of the tests. It was already a big hunt to find all of those now, adding even more does not seem like a very good idea.